components/Button/Button.module.scss

```diff
@@ -0,0 +1,112 @@
+@import "styles/variables/colors.module.scss";
+
+.common {
+  border-radius: 1.825rem;
+  border: solid 2px transparent;
+  box-sizing: border-box;
+  font-weight: 400;
+  cursor: pointer;
+}
+
+$large-size: 3.125rem;
+$medium-size: 2.5rem;
+$small-size: 1.825rem;
+
+@mixin primary($color, $colorDark, $colorDarker, $colorLight, $colorLighter, $textColor: $white) {
+  background: $color;
+  color: $textColor;
+
+  &:hover {
+    background: $colorLight;
+  }
+
+  &:focus {
+    box-shadow: 0px 0px 0px 2px $colorLighter, 0px 0px 8px 2px rgba($colorLighter, 0.5);
+  }
+
+  &:active {
+    background: $colorDark;
+    border-color: $color;
+  }
+
+  &[disabled] {
+    background: $colorDarker;
+    border-color: $colorDark;
+    color: $color;
+    cursor: not-allowed;
+  }
+}
+
+@mixin secondary($color, $colorDark, $colorDarker, $colorDarkest, $colorLight, $colorLighter, $textColor: $white) {
+  border-color: $color;
+  background-color: $colorDarkest;
+  color: $textColor;
+
+  &:hover {
+    background-color: $colorDarker;
+  }
+
+  &:focus {
+    box-shadow: 0px 0px 0px 2px $colorLighter, 0px 0px 8px 2px rgba($colorLighter, 0.5);
+  }
+
+  &:active {
+    background-color: $color;
+  }
+
+  &[disabled] {
+    background: $colorDarkest;
+    border-color: $colorDark;
+    color: $color;
+    cursor: not-allowed;
+  }
+}
+
+.blue.primary {
+  @include primary($blue, $blue-dark, $blue-darkest, $blue-light, $blue-lighter);
```

Comment:
Added these mixins for now - If there's a way of programmatically altering a base color to get the various states, that would be even neater. But this will do for now.

I haven't added the text-only button styles yet but that should be another mixin like primary and secondary and we can apply the colors as per the themes.

Reply:
this is cool though! how does it work exactly? just allows you to use the `.blue.primary` .blue.secondary` classes?

Reply:
Yeah, in https://github.com/populist-vote/web/pull/29/files#diff-10cce77e0d77f137ba5e5c2eddb80bb7b9167288a1797c3ff40e1bc9aa4c4ab7R23-R32 the corresponding classes are applied based on the props.

Reply:
really nice solution